waspc/waspls/src/Wasp/LSP/Syntax.hs

```diff
@@ -28,6 +31,15 @@ lspPositionToOffset srcString (J.Position l c) =
    in -- We add 1 to the length of each line to make sure to count the newline
       sum (map ((+ 1) . length) linesBefore) + fromIntegral c
 
+-- | @lspRangeToSpan srcString range@ converts the @range@ into a 'SourceSpan',
+-- where the start and end offsets of the span are 0-based offsets from the
+-- start of @srcString@.
+lspRangeToSpan :: String -> J.Range -> SourceSpan
```

Comment:
What is still not established here is relationship of Range with the srcString.

So if I got this right, Range operates in "2D" -> rows, columns, where source code is split into lines by newlines. So we know starting line and column, and we know ending line and column.
Span operates in "1D", where code source is concatenated into one big line.

So to go from Range to Span, we need to have source code that Range is referring to.

So that source string -> that is source of the whole file that Range is referring to, right? Because while that might sound clear, one might think that is maybe the source that Range is capturing.

So I would explain that sourceString is a piece of code, that given Range is defined in reference to it (its start and end are counted from the start of that source), and the same goes for Span that is then returned.

I don't think I used the best wording above to explain this, but I think this is important to capture, this relationship between that source string and given Range.

Reply:
Btw we have no test for this?